python/lsst/ctrl/bps/quantum_clustering_funcs.py

```diff
@@ -305,6 +315,135 @@ def add_clusters_per_quantum(
+    _LOG.debug("partition_dims = %s", partition_dims)
+
+    return cluster_dims, equal_dims, partition_dims
+
+
+def partition_cluster_values(
+    cluster_config: BpsConfig, cluster_label: str, partition_values: set[Any], num_cluster_values: int
+) -> dict[Any, int]:
+    """Partition given values into appropriately sized chunks.
+
+    Parameters
+    ----------
+    cluster_config : `lsst.ctrl.bps.BpsConfig`
+        BPS configuration for specific cluster label.
+    cluster_label : `str`
+        Cluster label for which to add clusters.
+    partition_values : `set` [`~typing.Any`]
+        Values that would be internal to clusters to be used in partitioning.
+    num_cluster_values : `int`
+        Number of values used when doing main clustering.
+
+    Returns
+    -------
+    partition_key_to_id : `dict` [`~typing.Any`, `int`]
+        Mapping of a value to a partition id.
+
+    Raises
+    ------
+    KeyError
+        When missing one of the partitioning config values or when specifying
+        multiple sizing values.
+    RuntimeError
+        When number of cluster dimension values is larger than
+        partitionMaxClusters.
+    """
+    _, partition_max_clusters = cluster_config.search("partitionMaxClusters")
+    _, partition_max_size = cluster_config.search("partitionMaxSize")
+    _LOG.debug("partition_max_clusters = %s", partition_max_clusters)
+    _LOG.debug("partition_max_size = %s", partition_max_size)
+
+    if not partition_max_clusters and not partition_max_size:
+        raise KeyError(
+            f"Defined partitionDimensions for cluster {cluster_label}, but missing one of the following:"
+            "partitionMaxClusters or partitionMaxSize"
+        )
+    elif partition_max_clusters and partition_max_size:
+        raise KeyError(
+            f"Defined more than one of the following for cluster {cluster_label}: "
+            "partitionMaxClusters and partitionMaxSize"
+        )
+
+    partition_max_chunks = 0
+    if partition_max_clusters:
+        # User wants to define the total number of clusters for this label
+        # which translates into jobs.
+        if num_cluster_values > partition_max_clusters:
+            raise RuntimeError(
+                f"Cluster {cluster_label}: partitionMaxClusters ({partition_max_clusters}) "
+                f"must same or larger than the number of cluster dimension values ({num_cluster_values})"
+            )
+
+        partition_max_chunks = partition_max_clusters // num_cluster_values
+        _LOG.debug("max_chunks = %s", partition_max_chunks)
+
+    if partition_max_chunks:
+        max_size, remainder = divmod(len(partition_values), partition_max_chunks)
+    else:
+        max_size = partition_max_size
+        remainder = 0
+
+    _LOG.debug("max_size = %s, remainder = %s", max_size, remainder)
+
+    # Make partitions to be used across all clusters for this label.
```

Comment:
It would be nice if this comment could include more details how the partitioning work as the logic below is not easy to follow. Extra brownie points will be awarded  for providing an example like "values 0, 1, 2 , 3, 4 will be partitioned as follow if `max_size` is 2".